components/client/src/main/java/com/hotels/styx/client/netty/connectionpool/NettyToStyxResponsePropagator.java

```diff
@@ -120,20 +120,16 @@ private RuntimeException toStyxException(Throwable cause) {
     public void channelInactive(ChannelHandlerContext ctx) throws Exception {
         ensureContentProducerIsCreated(ctx);
 
+        TransportLostException cause = new TransportLostException(ctx.channel().remoteAddress(), origin);
+        scheduleResourcesTearDown(ctx, cause);
+        contentProducer.ifPresent(producer -> producer.channelInactive(cause));
+    }
+
+    private void scheduleResourcesTearDown(ChannelHandlerContext ctx, Throwable cause) {
         ctx.channel().eventLoop().schedule(
-                () -> contentProducer.ifPresent(producer -> producer.idleStateEvent(
-                        new ResponseTimeoutException(
-                                origin,
-                                "channelInactive",
-                                producer.receivedBytes(),
-                                producer.receivedChunks(),
-                                producer.emittedBytes(),
-                                producer.emittedChunks()))),
+                () -> contentProducer.ifPresent(producer -> producer.tearDownResources(cause)),
```

Comment:
The `tearDownResources` event is scheduled in a controlled fashion. That is, the `tearDownResources` is always scheduled after a short time to ensure state machine is orderly brought down. Therefore `cause` doesn't have to be passed in.

Reply:
I would say that if we have to publish an error, it will be more informative if it is the same error that caused the "channelInactive"  so they can be easily linked when reading the logs, that if we make up an Exception as in your example below: TimedOutExceptionWhatever. 
But I haven't seen the logs to be honest...